src/svcutilcore/src/Microsoft/Tools/ServiceModel/SvcUtil/ToolRuntime.cs

```diff
@@ -30,30 +30,14 @@ internal ToolExitCodes Run()
             {
                 InputModule inputModule = InputModule.LoadInputs(_options);
 
-                Tool.Assert(_options.GetToolMode().HasValue, System.SR.Format(System.SR.AmbiguousToolUseage, Options.Cmd.Target, Options.Cmd.Validate));
-                ToolMode toolMode = _options.GetToolMode().Value;
-
-                return ExecuteToolMode(toolMode, inputModule);
+                Tool.Assert(_options.GetToolMode().HasValue, System.SR.Format(System.SR.AmbiguousToolUseage));
+                return ExecuteToolMode(inputModule);
             }
         }
 
-        private ToolExitCodes ExecuteToolMode(ToolMode toolMode, InputModule inputModule)
+        private ToolExitCodes ExecuteToolMode(InputModule inputModule)
         {
-            switch (toolMode)
-            {
-                case ToolMode.XmlSerializerGeneration:
-                    GenerateSerializer(inputModule);
-                    break;
-                case ToolMode.DataContractImport:
-                case ToolMode.ProxyGeneration:
-                case ToolMode.ServiceContractGeneration:
-                case ToolMode.DataContractExport:
-                case ToolMode.Validate:
-                case ToolMode.MetadataFromAssembly:
-                case ToolMode.WSMetadataExchange:
-                    throw new System.ArgumentException(System.SR.ErrInvalidTarget);
-            }
-
+            GenerateSerializer(inputModule);
             return ToolExitCodes.Success;
```

Comment:
You should keep the switch statement or at least an if statement here. If the tool mode is DisplayHelp but we somehow get into this method, we will run GenerateSerializer instead of it being a noop.

Reply:
DisplayHelp won't invoke the code here. If the parameter contains help, it will set ToolMode as DisplayHelp and it will then be called before initiate the input model as below.

if (_options.GetToolMode() == ToolMode.DisplayHelp)
            {
                ToolConsole.WriteHelpText();
                return ToolExitCodes.Success;
            }
            else
            {
                InputModule inputModule = InputModule.LoadInputs(_options);

                Tool.Assert(_options.GetToolMode().HasValue, System.SR.Format(System.SR.AmbiguousToolUseage));
                return ExecuteToolMode(inputModule);
            }